packages/design-system/src/components/OcResourceIcon/OcResourceIcon.vue

```diff
@@ -92,12 +87,9 @@ export default defineComponent({
         return defaultFolderIcon
       }
 
-      let icon =
-        (iconMapping[unref(extension)] as IconType) ||
-        iconMappingInjection?.mimeType[unref(mimeType)] ||
-        iconMappingInjection?.extension[unref(extension)]
+      let icon = resolveFileIcon(unref(extension))
```

Comment:
a) can be a const
b) you dropped all the injection now, right?! that's the mechanism that let's viewer/editor apps register icons for file types (e.g. the gpx-viewer app).

Reply:
a) true, thx!
b) there's still some mapping in `announceApplicationsReady` which I figured would be sufficient for viewer/editor apps at first glance, but I haven't really investigated tbh